### PR TITLE
Charm details relations

### PR DESCRIPTION
**Pull request by huwshimi**
Added charm relations to the details page. Added search results for relations.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- view the details for a charm
- on the right hand side of the readme there should be a list of relations
- clicking on a relation should show search results for that relation

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3686/
Test PASSed.

File: jujugui/static/gui/src/app/components/entity-content/entity-content.js [hunk 1]
```diff
@@ -156,6 +156,11 @@ YUI.add('entity-content', function() {
                   getFile={this.props.getFile} />
               </div>
               <div className="four-col">
+                <juju.components.EntityContentRelations
+                  changeState={this.props.changeState}
+                  relations={this.props.entityModel.get('relations')} />
+              </div>
+              <div className="four-col">
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/entity-content/entity-content.js]: I don't really like how these old styles are bleeding in :)

File: jujugui/static/gui/src/app/components/entity-content/test-entity-content-relations.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    var expected = (
+      <div className="section" id="relations">
+        <h3 className="section__title">
+          Relations
+        </h3>
+        <ul className="section__list">
+          <li className="link section__list-item"
+            role="button"
+            tabIndex="0"
+            onClick={output.props.children[1].props.children[0].props.onClick}
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/entity-content/test-entity-content-relations.js]: instance trick?

**Comment by hatched** [comment]
Looks awesome! :+1: QA :ok:

File: jujugui/static/gui/src/app/components/entity-content/test-entity-content-relations.js (hunk 2 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/entity-content/test-entity-content-relations.js]: I can't do that here as the values are bound to the called method, and the instance will be different in the tests.

**Comment by kadams54** [comment]
Nice work! :+1:

**Comment by hatched** [comment]
Thanks Huw :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**